flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -70,77 +80,111 @@ public static String toHiveTypeName(DataType type) {
 	public static TypeInfo toHiveTypeInfo(DataType type) {
 		checkNotNull(type, "type cannot be null");
 
-		if (type.equals(DataTypes.BOOLEAN())) {
-			return TypeInfoFactory.booleanTypeInfo;
-		} else if (type.equals(DataTypes.TINYINT())) {
-			return TypeInfoFactory.byteTypeInfo;
-		} else if (type.equals(DataTypes.SMALLINT())) {
-			return TypeInfoFactory.shortTypeInfo;
-		} else if (type.equals(DataTypes.INT())) {
-			return TypeInfoFactory.intTypeInfo;
-		} else if (type.equals(DataTypes.BIGINT())) {
-			return TypeInfoFactory.longTypeInfo;
-		} else if (type.equals(DataTypes.FLOAT())) {
-			return TypeInfoFactory.floatTypeInfo;
-		} else if (type.equals(DataTypes.DOUBLE())) {
-			return TypeInfoFactory.doubleTypeInfo;
-		} else if (type.equals(DataTypes.STRING())) {
-			return TypeInfoFactory.stringTypeInfo;
-		} else if (type.equals(DataTypes.DATE())) {
-			return TypeInfoFactory.dateTypeInfo;
-		} else if (type.equals(DataTypes.BYTES())) {
-			return TypeInfoFactory.binaryTypeInfo;
-		} else if (type.equals(DataTypes.TIMESTAMP())) {
-			return TypeInfoFactory.timestampTypeInfo;
-		} else {
-			LogicalType logicalType = type.getLogicalType();
-			Class clazz = logicalType.getClass();
-
-			if (clazz == BinaryType.class || clazz == VarBinaryType.class) {
-				// Hive doesn't support variable-length binary string
+		if (type instanceof AtomicDataType) {
```

Comment:
note that the part in this `if` clause doesn't change logic and is only added one tab